core/src/main/scala/scalaz/Kleisli.scala

```diff
@@ -244,6 +244,22 @@ abstract class KleisliInstances extends KleisliInstances0 {
     new KleisliCatchable[F, A] {
       implicit def F = F0
     }
+
+  implicit def kleisliNondeterminism[F[_]: Nondeterminism, E] = new Nondeterminism[Kleisli[F, E, ?]] {
```

Comment:
Also, tests for ambiguity in the presence of `F[_]: Nondeterminism` will be needed; see many neighbors to these tests in `KleisliTest.scala`.